### PR TITLE
Use SwiftHash

### PR DESCRIPTION
**Pull request by onmyway133**
- Use the homegrown SwiftHash, it contains just 1 file
- I've tested it a lot, even with https://github.com/minimaxir/big-list-of-naughty-strings
- This is not about file size, there's no problem with CryptoSwift, it's just about taste 😈

### CODE REVIEW

File: Source/Shared/Storage/DiskStorage.swift [hunk 1]
```diff
@@ -276,16 +276,7 @@ public final class DiskStorage: StorageAware {
...
   func fileName(_ key: String) -> String {
-    if let digest = key.data(using: String.Encoding.utf8)?.md5() {
-      var string = ""
-      for byte in digest {
-        string += String(format:"%02x", byte)
-      }
-
-      return string
-    } else {
-      return key.base64()
```
Comment (onmyway133) [inline Source/Shared/Storage/DiskStorage.swift]: If it sometimes can't `md5`, then we fallback to `base64`, which will cause inconsistent

**Comment by zenangst** [comment]
@onmyway133 you have some merge conflicts that you need to resolve here.

@vadymmarkov what do you think? Should we switch from CryptoSwift to SwiftHash? 😎

**Comment by vadymmarkov** [comment]
@zenangst @onmyway133 Could `CryptoSwift` be helpful if we implement encrypted cache storage?

**Comment by JohnSundell** [comment]
Using CryptoSwift also lets us take advantage of performance/bug fixes in a well-established, widely used library.

**Comment by zenangst** [comment]
Yeah after today's discussions it would probably be smart to stick with CryptoSwift!

**Comment by zenangst** [comment]
@hyperoslo/ios  I think we all agree that we don't want to go in this direction? Close PR?

**Comment by onmyway133** [comment]
😢

**Comment by zenangst** [comment]
Sorry @onmyway133, these things happen *hugs*

**Pull request closed without merging**